nimble/controller/src/ble_ll_hci_vs.c

```diff
@@ -356,6 +356,33 @@ ble_ll_hci_vs_set_local_irk(uint16_t ocf, const uint8_t *cmdbuf, uint8_t cmdlen,
 }
 #endif
 
+#if MYNEWT_VAL(BLE_LL_HCI_VS_SET_SCAN_CFG)
+static int
+ble_ll_hci_vs_set_scan_cfg(uint16_t ocf, const uint8_t *cmdbuf, uint8_t cmdlen,
+                           uint8_t *rspbuf, uint8_t *rsplen)
+{
+    const struct ble_hci_vs_set_scan_cfg_cp *cmd = (const void *)cmdbuf;
+    int rc;
+
+    if (cmdlen != sizeof(*cmd)) {
+        return BLE_ERR_INV_HCI_CMD_PARMS;
+    }
+
+    if (cmd->flags > BLE_HCI_OCF_VS_SET_SCAN_CFG_NO_EXT) {
```

Comment:
this should be converted from le32 to host order before check (even if this will be noop on most systems)

it should (at least for now) disallow command if both types are to be disabled